milli/src/search/query_tree.rs

```diff
@@ -313,7 +313,7 @@ pub struct TypoConfig<'a> {
 
 /// Return the `QueryKind` of a word depending on `authorize_typos`
 /// and the provided word length.
-fn typos<'a>(word: String, authorize_typos: bool, config: TypoConfig<'a>) -> QueryKind {
+fn typos(word: String, authorize_typos: bool, config: TypoConfig<'_>) -> QueryKind {
```

Comment:
Is this anonymous lifetime necessary?
```suggestion
fn typos(word: String, authorize_typos: bool, config: TypoConfig) -> QueryKind {
```

Reply:
This one is not needed :tada:

Reply:
fixed